### ISSUE TITLE
Make ggplotly respect override.aes in guide_legend

### ISSUE THREAD

**Issue by talgalili** [ropensci/plotly#989]
A short example:

```r
library(ggplot2)
p <- ggplot(economics_long, aes(date, value01, colour = variable)) +
  geom_line(size = .5) + 
  guides(colour = guide_legend(override.aes = list(size = 2)))
p
```

Notice the thicker lines in the legend on the right

![image](https://cloud.githubusercontent.com/assets/976006/25766422/201e4bec-31fb-11e7-80af-b7fe962f1aeb.png)

However, this new element is ignored in ggplotly:

```r
ggplotly(p)
```

![image](https://cloud.githubusercontent.com/assets/976006/25766432/30ac5c10-31fb-11e7-950e-753d4f6b54e2.png)

This becomes even more critical when using a very low level of alpha, and then needing to adjust it for the legend.

Thanks.

**Comment by DarioS** [comment]
Similarly, if a legend is modified with `scale_colour_manual`, plotly discards most of the modifications and plots more or less the original one.

```
library(plotly)
dataTable <- data.frame(index = 1:10, group = rep(c("low", "undetermined", "high"), c(4, 2, 4)),
                        value = c(7.6, 8.8, 9.0, 9.4, 5.6, 8.3, 3.3, 4.5, 5.2, 3.0))
aPlot <- ggplot(dataTable, aes(x = index, y = value)) + geom_point(aes(colour = group))
legendStyle <- scale_colour_manual(name = "Group", breaks = c("low", "high"),
               values = setNames(c("blue", "black", "red"), c("low", "undetermined", "high")),
	       labels = c("Low (n = 4)", "High (n = 4)"))
aPlot <- aPlot + legendStyle
aPlot # As desired.
ggplotly(aPlot) # Loss of legend modifications.
```

It uses the updated name, however.

**Comment by antuki** [comment]
Sharing this need too. Such a shame that plotly does not allow overrinding aes for legend. It would be useful, especially when you want to modify the aspect  (line type, opacity) of one of the markers but not the other of the same legendgroup. I did not manage to do it, even when modifying manually the plotly object and I do not think it is possible with plotly at all... But I really hope it will be one day ! Many thanks.

**Comment by farshadf** [comment]
Is this issue still open? We have encountered the same problem, and it is frustrating. Hope somebody does something about it soon.

**Comment by ShinyFabio** [comment]
After 5 years is there a solution?

**Comment by DarioS** [comment]
No, but I see that the package continues to be maintained and new versions released regularly. I still hold out hope.

**Comment by velicknd** [comment]
Also encountering this issue. A fix for it would be fantastic!

**Comment by danalvarez** [comment]
Anyone have any workarounds for this?

**Comment by talgalili** [comment]
No solution yet.
Although with having LLM these days, maybe someone could try asking chatgpt to solve it...
